src/net_packet.c

```diff
@@ -409,12 +411,22 @@ bool send_sptps_data(void *handle, uint8_t type, const void *data, size_t len) {
 			if(to->mtu >= len) {
 				to->mtu = len - 1;
 			}
+
+			if(to->mtu && (to->mtuprobes > 30) && !type) {
+				to->mtuprobes = 1;
+			}
 		} else {
 			logger(mesh, MESHLINK_WARNING, "Error sending UDP SPTPS packet to %s: %s", to->name, sockstrerror(sockerrno));
 			return false;
 		}
 	}
 
+	if(to->mtu && (to->mtuprobes > 30) && !type) {
```

Comment:
This doesn't feel right. We should not need to reset the ping timer because we are trying to send packets. This would also mean that when we are sending data frequently (more than once every ping interval), we keep resetting the timer before it ever expires, and then the MTU handlers will not run.

Reply:
One of our optimizations that we planned in PMTU probing process is that If there are some data packets being sent to the other node at the end then those packets are also keeping NAT mappings alive and so we can skip the next ping probe(s) and done by resetting MTU time handler with `mesh->pinginterval` value.